internal/provider/kubernetes/controller.go

```diff
@@ -1481,7 +1481,7 @@ func (r *gatewayAPIReconciler) processGateways(ctx context.Context, managedGC *g
 				// processGatewayParamsRef has been called for this gateway, its EnvoyProxy should exist in resourceTree
 				ep = resourceTree.GetEnvoyProxy(gtw.Namespace, gtw.Spec.Infrastructure.ParametersRef.Name)
 			}
-			r.processServiceClusterForGateway(ep, &gtw, resourceMap)
+			r.processServiceClusterForGateway(ep, &gtw, resourceMap, resourceTree)
```

Comment:
we can do the check at L1483 instead of processServiceClusterForGateway

Reply:
I think the check logic is a part of the `processServiceClusterForGateway`

Reply:
the name called `processServiceClusterForGateway`, so we'd need do the check before call it.
there's another one called `processServiceClusterForGatewayClass`.

Reply:
ok, deal with this after #6847 and add the release note for this bug